yt/python/yt/wrapper/py_wrapper.py

```diff
@@ -756,3 +759,184 @@ def with_formats(func, input_format=None, output_format=None):
     if output_format is not None:
         _set_attribute(func, "output_format", output_format)
     return func
+
+
+class DockerRespawner:
+    def __init__(
+            self,
+            image,
+            target_platform,
+            docker,
+            python,
+            env=None,
+            main_scipt_path=None,
+            cwd=None,
+            homedir=None,
+            python_lib_paths=None,
+            mount=None,
+            need_sudo=False,
+    ):
+        self._image = image
+        self._platform = target_platform
+        self._docker = docker
+        self._env = env
+        if self._env is None:
+            # by default, we forward only YT-specific variables
+            self._env = {
+                key: value
+                for key, value in os.environ.items()
+                if key.startswith('YT_')
+            }
+        self._main_script_path = main_scipt_path if main_scipt_path is not None else os.path.abspath(sys.argv[0])
+        self._cwd = cwd if cwd is not None else os.path.abspath(os.getcwd())
+        self._homedir = homedir if homedir is not None else os.path.expanduser("~")
+        # we need to provide all python libraries to a container to ensure picking works
+        self._python_lib_paths = (
+            python_lib_paths
+            if python_lib_paths is not None else
+            [os.path.abspath(path) for path in sys.path]
+        )
+        self._mount = mount
+        self._need_sudo = need_sudo
+
+    def _make_docker_env_args(self, env):
+        return list(itertools.chain(*[
+            ("-e", "{0}={1}".format(key, value))
```

Comment:
Is it safe to pass env variables as `"{0}={1}"`? What if it contains white space or some other special characters?

Reply:
It is a very good point.
There is a similar problem not only with environment variables, but also with other values (e.g. mount points).

So I'll add escaping for the generated command.